ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -86,12 +86,24 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
   const visual = merge({}, DEFAULT_VISUAL, props.spec.visual);
 
   // convert Perses dashboard format to be ECharts compatible
-  const yAxis = {
+  const yAxis: YAXisComponentOption = {
     show: y_axis?.show ?? DEFAULT_Y_AXIS.show,
-    min: y_axis?.min, // leaves min and max undefined by default to let ECharts calcualate
+    min: y_axis?.min,
     max: y_axis?.max,
   };
 
+  if (yAxis.min === undefined) {
+    // sets minimum axis label relative to data instead of zero
+    yAxis.min = (value) => {
+      // https://echarts.apache.org/en/option.html#yAxis.min
+      if (value.min <= 1) {
+        return 0;
+      }
+      // allows for padding between origin and first series
+      return value.min * 0.8;
+    };
+  }
+
```

Comment:
[meandering thought about this component] There's a lot of utilities inside this component that make decisions about processing data and setting config. I'm wondering if it might be helpful to pull some of these out into utility functions that the component calls. It'd probably make the code a little more readable, it would make it easier to write some simple tests for the logic, and it might make it easier for us to eventually make some of that logic more accessible for embeddability with the `LineChart`. 

[specific, optional ask for this PR] Not asking you to pull them all out in this PR, but maybe it would be useful to pull out the bits you touched, so you can add a simple unit test. Thoughts?